sdk/go/common/util/maputil/sorted.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2016-2020, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package maputil
+
+import (
+	"cmp"
+	"slices"
+)
+
+func SortedKeys[K cmp.Ordered, V any](m map[K]V) []K {
+	keys := make([]K, 0)
```

Comment:
Would `make([]K, len(m))` work here? And if so, is it worth it in terms of allocations? Not an expert.

Reply:
Yes and yes, or `make([]K, 0, len(m))` if you wanted to keep the append rather than `keys[i]` style below

Reply:
Also this changes to not use prealloc which means it will always return an allocated slice rather than nil now. That _might_ matter for some code paths.